### PR TITLE
SAM-73 Filter samplesets dynamic dropdown

### PR DESCRIPTION
**Pull request by charleshtrenholm**
* Changes free text input of fields to filter samples on to dynamic dropdown
* Creates get_sampleset_meta method that returns list of sample columns with which to populate dynamic dropdown
* updates spec.json to use new method of getting filter fields based on other param methods in a kbase app/method

### CODE REVIEW

File: test/sample_set_filter_test.py
Comment (Tianhao-Gu) [inline test/sample_set_filter_test.py:228]: I am not sure what search API return looks like. Something like `['isgn', 'bar', 'foo'...]`? Can we do assertCountEqual to example the whole return?

File: test/sample_set_filter_test.py
Comment (Tianhao-Gu) [inline test/sample_set_filter_test.py:225]: missing `ctx` argument. Also I think `sample_set_ref` should be a list argument (e.g. [self.sample_set_2_ref])

File: test/sample_set_filter_test.py
Comment (Tianhao-Gu) [inline test/sample_set_filter_test.py:227]: here ret is `[results]`. I think you mean `self.assertEqual(type(ret[0]), list)` and `self.assertEqual(type(ret[0][0]), str)` here?

File: sample_uploader.spec
Comment (Tianhao-Gu) [inline sample_uploader.spec:214]: Ok, but I would name it as `sampleset_refs`. It's your call.

File: sample_uploader.spec
Comment (Tianhao-Gu) [inline sample_uploader.spec:209]: `a given sampleset` --> `given samplesets`

File: lib/sample_uploader/sample_uploaderImpl.py
Comment (Tianhao-Gu) [inline lib/sample_uploader/sample_uploaderImpl.py:788]: `sample_search_api` client needs to be reinstalled so that `get_sampleset_meta` is available. Currently, only `filter_samples` is callable.

File: lib/sample_uploader/sample_uploaderImpl.py
Comment (Tianhao-Gu) [inline lib/sample_uploader/sample_uploaderImpl.py:786]: Should be probably fine. But I would call `sample.get('id') and sample.get('version')` here. Or better do a check here before calling the search API. Because `sample_set_ref` can theoretically be any object type. If objects other than `SampleSet` were given, this probably will raise a KeyError.

**Review by Tianhao-Gu — CHANGES_REQUESTED** [review]
overall looks good. But a few things needed to be changed. local tests are also failing.

File: lib/sample_uploader/sample_uploaderImpl.py
Comment (charleshtrenholm) [reply lib/sample_uploader/sample_uploaderImpl.py:788]: I'm assuming I can't reinstall the client until the new method is released?

Comment (Tianhao-Gu) [reply lib/sample_uploader/sample_uploaderImpl.py:788]: You are right. Tests won't pass until the new method is released. We can revisit this PR once the new method is released. You can still install the client by running `kb-sdk install https://raw.githubusercontent.com/kbase/sample_search_api/master/sample_search_api.spec`

Comment (charleshtrenholm) [reply lib/sample_uploader/sample_uploaderImpl.py:788]: sample_search_api is already installed for the filter_samplesets method, so I think it would make sense to wait until a release is out that has get_sampleset_meta

File: test/sample_set_filter_test.py
Comment (charleshtrenholm) [reply test/sample_set_filter_test.py]: I tried to mimic the results based off of what I can see from sample_service_validator_config, but I wasn't sure what in the test csv was a custom field or not, so this test may need some tweaking

**Pull request merged**
